### PR TITLE
Added Sort and Filter to the API layer and to the UI

### PR DESCRIPTION
**Pull request by jordaniza**
Created the backend logic to allow sorting and filtering by Reward on the backend. 

Changes are mainly located in the `packages/react-app/src/pages/api/bounties/index.ts` file. I decided to split Filter and Sort functionality, so that future maintainers can manage them independently.

You can test the changes using postman or by hitting the api directly, some sample queries:

```
# status Open, no text filtering, at least 1000 bank reward in descending order
http://localhost:3000/api/bounties?status=Open&search&gte=1000&sortBy=reward&asc=0

# status Completed, no text filtering, less than 1000 bank reward
http://localhost:3000/api/bounties?status=Completed&search&lte=1000

# status all, no text filtering, at least 1000 bank reward and up to 4000 bank
http://localhost:3000/api/bounties?status=&search&gte=1000&lte=4000&sortBy=reward
```

Changelog:

- Added sample queries to the mongo/bounties/bounty-queries.js
- Created type declaration files for accepted query filters and sorts
- handleFilters function was getting very large by the end, so I split that up into separate functions.
- Added less than or equal to, greater than or equal to and sort functionality to the query

### CODE REVIEW

**Comment by jordaniza** [comment]
Just as a comment, text search works but if seeding the DB you need to create an index. I should add that to the onboarding documentation.

**Comment by jordaniza** [comment]
Added the React changes, you can now:

* Sort by reward only
* Sort ascending and descending
* Sort by a range

Using the UI

@amaredeus the latest 2 commits contain the UI changes, building on the initial API changes

File: packages/react-app/src/components/pages/Bounties/index.tsx [hunk 1]
```diff
@@ -22,6 +22,11 @@ const Bounties = ({ id }: PreFilterProps): JSX.Element => {
 	const [page, setPage] = useState(0);
 	const [status, setStatus] = useState('');
 	const [search, setSearch] = useState('');
+	const [gte, setGte] = useState(0);
+	// how to handle the lte === 0 case?
+	const [lte, setLte] = useState(Infinity);
```
Comment (Oktalize) [inline packages/react-app/src/components/pages/Bounties/index.tsx:27]: can we add the min/max state variables to the `useEffect` that `setPage(0)` on line 58? that way when they change the pagination doesn't weird out?

Comment (jordaniza) [reply packages/react-app/src/components/pages/Bounties/index.tsx:27]: @Oktalize Thank you for reviewing and finding that. On the case for pagination.

**Pull request merged**